v03_pipeline/lib/reference_data/gencode/mapping_gene_ids.py

```diff
@@ -33,115 +20,45 @@
     'phase',
     'info',
 ]
+EXPECTED_ENSEMBLE_TO_REFSEQ_FIELDS = 3
 
 
-def _get_pickle_file(path):
-    root, ext = os.path.splitext(path)
-    return root + '.pickle'
-
-
-def _load_parsed_data_or_download(gencode_release, download_path):
-    gene_id_mapping = {}
+def load_gencode_gene_symbol_to_gene_id(gencode_release: int) -> dict[str, str]:
     url = GENCODE_GTF_URL.format(gencode_release=gencode_release)
-    gencode_gtf_path = os.path.join(download_path, os.path.basename(url))
-    pickle_file = _get_pickle_file(gencode_gtf_path)
-    if path_exists(pickle_file):
-        logger.info(
-            'Use the existing pickle file {}.\nIf you want to reload the data, please delete it and re-run the data loading.'.format(
-                pickle_file,
-            ),
-        )
-        if is_gs_path(pickle_file):
-            p = pickle.loads(stream_gs_file(pickle_file))
-        else:
-            with open(pickle_file, 'rb') as handle:
-                p = pickle.load(handle)
-        gene_id_mapping.update(p)
-    elif not path_exists(gencode_gtf_path):
-        gencode_gtf_path = download_file(url, to_dir=download_path)
-        logger.info(f'Downloaded to {gencode_gtf_path}')
-    else:
-        logger.info(
-            'Use the existing downloaded file {}.\nIf you want to re-download it, please delete the file and re-run the pipeline.'.format(
-                gencode_gtf_path,
-            ),
-        )
-
-    return gene_id_mapping, gencode_gtf_path
-
-
-def _parse_gtf_data(gencode_gtf_path):
-    gene_id_mapping = {}
-    logger.info(f'Loading {gencode_gtf_path}')
-    is_gs = is_gs_path(gencode_gtf_path)
-    gencode_file = (
-        gzip.decompress(stream_gs_file(gencode_gtf_path, raw_download=True))
-        .decode()
-        .split('\n')
-        if is_gs
-        else gzip.open(gencode_gtf_path, 'rt')
-    )
-    for i, line in enumerate(gencode_file):
-        line = line.rstrip('\r\n')
+    response = requests.get(url, stream=True, timeout=10)
+    gene_symbol_to_gene_id = {}
+    for line in gzip.GzipFile(fileobj=response.raw):
+        line = line.decode('ascii')  # noqa: PLW2901
```

Comment:
nice, this became very simple. is there a downside to taking the pickle file part out of the download process? do you know why it was there in the first place?